### PR TITLE
Add function to drop the osm chunk

### PR DESCRIPTION
**Pull request by konskov**
requires PR 6815

### CODE REVIEW

File: src/chunk.c [hunk 1]
```diff
@@ -5189,3 +5190,42 @@ get_chunks_in_creation_time_range(Hypertable *ht, int64 older_than, int64 newer_
...
+				 errmsg("OSM chunk %s.%s does not correspond to hypertable %s.%s",
+						chunk_schema_name,
+						chunk_table_name,
+						ht->fd.schema_name.data,
+						ht->fd.table_name.data)));
+	ts_chunk_validate_chunk_status_for_operation(ch, CHUNK_DROP, true /*throw_error */);
+	/* do not drop any chunk dependencies */
+	ts_chunk_drop(ch, DROP_RESTRICT, LOG);
+	/* reset hypertable OSM status */
+	ht->fd.status = HYPERTABLE_STATUS_DEFAULT;
```
Comment (gayyappan) [inline src/chunk.c:5227]: should this be done via hypertable_update_status_osm call?

Comment (konskov) [reply src/chunk.c:5227]: yes, using `hypertable_update_status_osm` would probably be a better option as it takes the proper locks for updating the `timescaledb_catalog.hypertable` row

File: sql/chunk.sql [hunk 2]
```diff
@@ -75,3 +75,9 @@ CREATE OR REPLACE FUNCTION _timescaledb_functions.attach_osm_table_chunk(
    hypertable REGCLASS,
    chunk REGCLASS)
 RETURNS BOOL AS '@MODULE_PATHNAME@', 'ts_chunk_attach_osm_table_chunk' LANGUAGE C VOLATILE;
+
+-- internal API used by OSM extension to drop an OSM chunk table from the hypertable
+CREATE OR REPLACE FUNCTION _timescaledb_functions.drop_osm_table_chunk(
+   hypertable REGCLASS,
+   chunk REGCLASS)
+RETURNS BOOL AS '@MODULE_PATHNAME@', 'ts_chunk_drop_osm_table_chunk' LANGUAGE C VOLATILE;
```
Comment (fabriziomello) [inline sql/chunk.sql:83]: Why not pass only the chunk regclass to this API??

**Comment by erimatnor** [comment]
Seems to be stalled, so closing for now.

**Comment by gayyappan** [comment]
newer PR: https://github.com/timescale/timescaledb/pull/7209

**Pull request closed without merging**